common/cross_sockets/XSocketServer.cpp

```diff
@@ -42,19 +42,26 @@ bool XSocketServer::init_server() {
     return false;
   }
 
+  int yes = 1;
+
 #ifdef OS_POSIX
-  int server_socket_opt = SO_REUSEADDR | SO_REUSEPORT;
+  if (set_socket_option(listening_socket, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(yes)) < 0) {
+    close_server_socket();
+    return false;
+  }
+  // macOS doesn't support setting multiple options at once, so we have to do this separately.
+  if (set_socket_option(listening_socket, SOL_SOCKET, SO_REUSEPORT, &yes, sizeof(yes)) < 0) {
+    close_server_socket();
+    return false;
+  }
```

Comment:
I figured Linux can handle setting them separately, so they can just have the same code.

But I can create separate `__APPLE__` and `__linux` sections.

Reply:
This seems fine, better to have only 1 implementation I think